src/aapns/api.py

```diff
@@ -0,0 +1,134 @@
+import json
+import ssl
+from asyncio import get_event_loop, wait_for, Lock
+from typing import Optional
+
+from structlog import BoundLogger
+
+from . import errors, config, models, connection
+
+
+
+class APNS:
+    def __init__(self, client_cert_path, server, logger, ssl_context, *, auto_reconnect=False, timeout=None):
+        self.client_cert_path = client_cert_path
+        self.server = server
+        self.logger = logger
+        self.ssl_context = ssl_context
+        self.auto_reconnect = auto_reconnect
+        if timeout is None:
+            self.run_with_timeout = self.run_without_timeout
```

Comment:
Nitpick.

Maybe it's better to use `self.run` (or other name) then
```
if timeout is None:
    self.run = self.run_without_timeout
else:
    self.run = self.run_with_timeout
```

then the subsequent call will be not `self.run_with_timeout` while it's actually running without timeout